File: 1ºAno/Lista3/C/Casa/files/ex1.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main() {
    
    for (;;) {
        char *dados = (char*)malloc(10 * sizeof (char));
        int vals[3] = {0, 0, 0};
        for (;;) {
            printf ("Na mesma linha, entre com os valores inteiros, respectivamente, referentes a:\n\n1) 1º Número da sequência (a1)\n2) Último número da sequência (an)\n3) Razão da sequência (r)\nPara sair, entre com 0 em todos os valores\n\n");
            fgets (dados, 10, stdin);
            char *pt = strtok(dados, " ");
            int i = 0;
            while (pt != NULL) {
                vals[i] = atoi(pt);
                pt = strtok(NULL, " ");
                ++i;
            }
            if (vals[0] == 0 && vals[1] == 0 && vals[2] == 0) {
                break;
            }
            else if (vals[2] == 0) {
                printf ("*Número(s) inválido(s)... Tente novamente*");
            }
            else {
                break;
            }
        }
        if (vals[0] == 0 && vals[1] == 0 && vals[2] == 0) {
            break;
        }
        printf ("Aqui estão os números de %d até %d, de %d em %d:\n", vals[0], vals[1], vals[2], vals[2]);
        if (vals[0] <= vals[1]) {
            while (vals[0] <= vals[1]) {
                printf ("%d ", vals[0]);
                vals[0] += vals[2];
            }
        }
        else {
            while (vals[0] >= vals[1]) {
                printf ("%d ", vals[0]);
                vals[0] -= vals[2];
            }
        }
        printf ("\n");
    }
    printf ("\nAdeus!");
    return 0;
}

File: 1ºAno/Lista3/Python/Casa/files/ex1.py
while True:
    while True:
        try:
            vals = input("Na mesma linha, entre com os valores, respectivamente, referentes a:\n\n1) 1º termo da sequência (a1)\n2) Último termo da sequência (an)\n3) Razão da sequência (r)\nPara sair, entre com 0 em todos os valores\n\n").split()
            vals[0], vals[1], vals[2] = int(vals[0]), int(vals[1]), int(vals[2])
        except:
            print ("*Número(s) inválido(s)... Tente novamente*")
        else:
            break
    if vals[0] == vals[1] == vals[2] == 0:
        break
    print (f"Aqui esão os números de {vals[0]} até {vals[1]}, de {vals[2]} em {vals[2]}:")
    if vals[0] <= vals[1]:
        while vals[0] <= vals[1]:
            print (f"{vals[0]} ", end="")
            vals[0] += vals[2]
    else:
        while vals[0] >= vals[1]:
            print (f"{vals[0]} ", end="")
            vals[0] -= vals[2]
    print ()